cmd/mqtt/main.go

```diff
@@ -122,6 +130,34 @@ func main() {
 		log.Fatalf(err.Error())
 	}
 
+	if cfg.mqttTargetHealthCheck != "" {
+		for i := 1; i <= maxHealthCheckRetries; i++ {
+			res, err := http.Get(cfg.mqttTargetHealthCheck)
+			if err != nil {
+				logger.Info(fmt.Sprintf("Broker not ready: %s ", err.Error()))
+			}
+			if res != nil {
+				body, err := ioutil.ReadAll(res.Body)
+				if err != nil {
+					logger.Warn(fmt.Sprintf("Error reading response body: %s", err.Error()))
+				}
+				if err := res.Body.Close(); err != nil {
+					logger.Warn(fmt.Sprintf("Error closing response body: %s", err.Error()))
+				}
+				if res.StatusCode == http.StatusOK {
+					logger.Info("MQTT Broker health check successful")
+					break
+				}
+				logger.Info(fmt.Sprintf("Broker not ready, status code: %d, body: %s", res.StatusCode, body))
+			}
+			if i == maxHealthCheckRetries {
```

Comment:
Should be probably moved to the top of the loop.

Reply:
> Yes, this is exactly what I meant - that the rest of the operations in this loop body should be skipped. Because if you can not do HTTP GET, do you really need to check the result? And do you really need to read HTTP body (remember that GET failed)? And then do you need to close the Body (remember, GET failed, and you did not read anything)?
Rest of code is already skipped - except sleep.
All operation you mention here is in `if res != nil` where `res` is result of `http.Get`. If that fails (and that is normal because adapter will be up before broker) then sleep and retry again.

Reply:
> Should be probably moved to the top of the loop.

Moving it to the top would do unnecessary sleep in situation when success is in firts iteration.

Reply:
My question was - why are you even checking `res` is you already had an `err` in `http.Get(cfg.mqttTargetHealthCheck)`? It's  like hoping that `res` will somehow be valid, although there was an error in the GET function itself.